src/napari_imagej/widget.py

```diff
@@ -0,0 +1,200 @@
+"""
+This module is an example of a barebones function plugin for napari
+
+It implements the ``napari_experimental_provide_function`` hook specification.
+see: https://napari.org/docs/dev/plugins/hook_specifications.html
+
+Replace code below according to your needs.
+"""
+from typing import Any, List
+from napari import Viewer
+from qtpy.QtWidgets import (
+    QWidget,
+    QHBoxLayout,
+    QVBoxLayout,
+    QPushButton,
+    QLineEdit,
+    QTableWidget,
+    QAbstractItemView,
+    QHeaderView,
+    QTableWidgetItem,
+    QLabel,
+)
+from scyjava import when_jvm_starts
+from napari_imagej.setup_imagej import ij, logger, java_import
+from napari_imagej._module_utils import functionify_module_execution
+from napari_imagej._napari_converters import init_napari_converters
+from threading import Thread
+
+
+class ImageJWidget(QWidget):
+    """The top-level ImageJ widget for napari."""
+    def __init__(self, napari_viewer: Viewer):
+        super().__init__()
+
+        # Install napari <-> java converters
+        when_jvm_starts(lambda: init_napari_converters())
+
+        self.viewer = napari_viewer
+        
+        self.setLayout(QVBoxLayout())
+
+        # Search Bar
+        self._search_widget: QWidget = QWidget()
+        self._search_widget.setLayout(QHBoxLayout())
+        self._searchbar: QLineEdit  = self._generate_searchbar()
+        self._search_widget.layout().addWidget(self._searchbar)
+
+        self.layout().addWidget(self._search_widget)
+
+        # Searcher plugins
+        def init_searchers(): 
+            self.searchers, self.resultConverters = self._generate_searchers()
+            self.searchService = self._generate_search_service()
+            # Enable the searchbar now that the searchers are ready
+            self._searchbar.setText("")
+            self._searchbar.setEnabled(True)
+        # Initialize the searchers, which will spin up an ImageJ gateway.
+        # By running this in a new thread,
+        # the GUI can be shown before the searchers are ready.
+        Thread(target=init_searchers).start()
+
+        # Results box
+        self.resultTableNames = [
+            "Modules:",
+            "Ops:"
+        ]
+        self.results = [[] for _ in self.resultTableNames]
+        self.resultTables, self.tableWidgets = self._generate_results_widget(self.resultTableNames)
+        self.layout().addWidget(self.resultTables)
+
+        # Module highlighter
+        self.focus_widget = QWidget()
+        self.focus_widget.setLayout(QVBoxLayout())
+        self.focused_module = None
+        self.focused_module_label = QLabel()
+        self.focus_widget.layout().addWidget(self.focused_module_label)
+        self.focused_module_label.setText("Display Module Here")
+        self.layout().addWidget(self.focus_widget)
+        self.focused_action_buttons = []  # type: ignore
```

Comment:
I'd like to split this construction up into two sections. The first section should create all of this widget's subwidgets in a logical order with good documentation.

The second should populate them in a separate thread.

Right now, these two tasks are combined into one, making it hard to read.